plugins/pluginmanagers/lazy.nix

```diff
@@ -0,0 +1,187 @@
+        pluginType =
+          either
+          package
+          (submodule {
+            options = {
+              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
+
+              pkg = mkOption {
+                type = package;
+                description = "Vim plugin to install";
+              };
+
+              name = helpers.mkNullOrOption str "Name of the plugin to install";
+
+              dev = helpers.defaultNullOpts.mkBool false "When true, a local plugin directory will be used instead. See config.dev";
+
+              lazy = helpers.defaultNullOpts.mkBool true "When true, the plugin will only be loaded when needed. Lazy-loaded plugins are automatically loaded when their Lua modules are required, or when one of the lazy-loading handlers triggers";
+
+              enabled = helpers.defaultNullOpts.mkBool true "When false then this plugin will not be included in the spec";
+
+              cond =
+                helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw bool)
+                "When false, or if the function returns false, then this plugin will not be loaded. Useful to disable some plugins in vscode, or firenvim for example.";
+
+              dependencies =
+                helpers.mkNullOrOption (helpers.nixvimTypes.eitherRecursive str listOfPlugins)
+                "Plugin dependencies";
+
+              init =
+                helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw bool)
+                "init functions are always executed during startup";
+
+              config =
+                helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw bool)
+                "config is executed when the plugin loads. The default implementation will automatically run require(MAIN).setup(opts). Lazy uses several heuristics to determine the plugin's MAIN module automatically based on the plugin's name. See also opts. To use the default implementation without opts set config to true.";
+
+              main = helpers.mkNullOrOption str "You can specify the main module to use for config() and opts(), in case it can not be determined automatically. See config()";
+
+              submodules = helpers.defaultNullOpts.mkBool true "When false, git submodules will not be fetched. Defaults to true";
+
+              event = with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str)))
+                "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+
+              cmd = with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str)))
+                "Lazy-load on command";
+
+              ft = with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str)))
+                "Lazy-load on filetype";
+
+              keys = with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str)))
```

Comment:
We could map the `LazyKeysSpec` submodule, but it's not really necessary because the user could do it with a `maybeRaw`

Reply:
How do you want to address this?

Reply:
Leave it like that, we'll see if we get complaints